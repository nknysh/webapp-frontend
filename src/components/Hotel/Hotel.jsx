import React, { useRef, useState } from 'react';
import { map, values } from 'ramda';
import { isNilOrEmpty } from 'ramda-adjunct';
import { useTranslation } from 'react-i18next';

import { propTypes, defaultProps } from './Hotel.props';
import {
  HotelDescription,
  HotelDetails,
  HotelGallery,
  HotelHighlight,
  HotelHighlights,
  HotelInfo,
  HotelName,
  HotelRegion,
  HotelStar,
  HotelStarRating,
  HotelStarText,
  MainSlider,
  NavSlider,
  StyledHotel,
} from './Hotel.styles';

import Rooms from '../../containers/Rooms';
import LinkButton from 'components/LinkButton';

// eslint-disable-next-line
const renderImage = ({ displayName, url }) => <img key={url} src={url} alt={displayName} />;
const renderFeature = value => <HotelHighlight key={value}>{value}</HotelHighlight>;

export const Hotel = ({
  id,
  amenities,
  description,
  name,
  photos,
  region,
  starRating,
  overview,
  highlights,
  ...props
}) => {
  const [showAmenities, setShowAmenities] = useState(false);
  const [showOverview, setShowOverview] = useState(false);
  const [seeHighlights, setSeeHighlights] = useState(false);
  const { t } = useTranslation();

  const sliderMain = useRef(null);
  const sliderNav = useRef(null);

  const renderedPhotos = values(map(renderImage, photos));

  return (
    <StyledHotel {...props}>
      <HotelDetails>
        {!isNilOrEmpty(photos) && (
          <HotelGallery>
            <MainSlider asNavFor={sliderNav} centerMode={false} fade={true} ref={sliderMain} slidesToShow={1}>
              {renderedPhotos}
            </MainSlider>
            <NavSlider
              asNavFor={sliderMain}
              centerMode={false}
              focusOnSelect={true}
              infinite={false}
              ref={sliderNav}
              slidesToShow={6}
              arrows={true}
            >
              {renderedPhotos}
            </NavSlider>
          </HotelGallery>
        )}
        <HotelInfo>
          <HotelStarRating>
            <HotelStar>star</HotelStar>{' '}
            <HotelStarText>
              {starRating} {t('star')}
            </HotelStarText>
          </HotelStarRating>
          <HotelName>{name}</HotelName>
          {region && <HotelRegion>{region}</HotelRegion>}
          <HotelDescription>{description}</HotelDescription>

          {!isNilOrEmpty(overview) && (
            <LinkButton
              aria-label={showOverview ? t('labels.hideOverview') : t('labels.seeOverview')}
              className="linkButton linkButton--in-list"
              onClick={() => setShowOverview(!showOverview)}
            >
              {showOverview ? `- ${t('labels.hideOverview')}` : `+ ${t('labels.seeOverview')}`}
            </LinkButton>
          )}
          {showOverview && <HotelHighlights>{map(renderFeature, overview)}</HotelHighlights>}

          {!isNilOrEmpty(highlights) && (
            <LinkButton
              aria-label={seeHighlights ? t('labels.hideHighlights') : t('labels.seeHighlights')}
              className="linkButton linkButton--in-list"
              onClick={() => setSeeHighlights(!seeHighlights)}
            >
              {seeHighlights ? `- ${t('labels.hideHighlights')}` : `+ ${t('labels.seeHighlights')}`}
            </LinkButton>
          )}
          {seeHighlights && <HotelHighlights>{map(renderFeature, highlights)}</HotelHighlights>}

          {!isNilOrEmpty(amenities) && (
            <LinkButton
              className="linkButton linkButton--in-list"
              aria-label={showAmenities ? t('labels.hideAmenities') : t('labels.seeAmenities')}
              onClick={() => setShowAmenities(!showAmenities)}
            >
              {showAmenities ? `- ${t('labels.hideAmenities')}` : `+ ${t('labels.seeAmenities')}`}
            </LinkButton>
          )}
          {showAmenities && <HotelHighlights>{map(renderFeature, amenities)}</HotelHighlights>}
        </HotelInfo>
        <Rooms hotelUuid={id} />
      </HotelDetails>
    </StyledHotel>
  );
};

Hotel.propTypes = propTypes;
Hotel.defaultProps = defaultProps;

export default Hotel;
