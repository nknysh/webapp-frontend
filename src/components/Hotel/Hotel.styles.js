import styled from 'styled-components';
import { Slider } from '@pure-escapes/webapp-ui-components';

import Card from 'components/Card';
import Rooms from 'containers/Rooms';

import {
  CardStarRating,
  CardStar,
  CardStarText,
  CardHighlights,
  CardHighlight,
} from 'containers/SearchResult/SearchResult.styles';

import { theme } from 'styles';

export const StyledHotel = styled.div`
  .linkButton {
    margin: 1rem 0;
  }
`;

export const HotelDetails = styled.div``;

export const HotelName = styled(Card.Title)`
  border-bottom: 0;
  margin: 0 0 ${theme.spacing.gutter / 2}px;
  padding: 0;
`;

export const HotelRegion = styled.h3`
  font-size: ${theme.fonts.sizes.normal}px;
  color: ${theme.palette.neutral};
  text-transform: uppercase;
  font-weight: ${theme.fonts.normal};
  margin: 0 0 ${theme.spacing.gutter}px;
  padding: 0;
`;

export const HotelStarRating = styled(CardStarRating)`
  border: 0;

  :last-child {
    margin-bottom: 0;
  }

  ${props => props.theme.breakpoints.tablet`
        margin: ${theme.spacing.gutter / 2}px 0 ${theme.spacing.gutter}px;
    `}
`;

export const HotelStar = styled(CardStar)`
  font-size: 20px !important;
`;

export const HotelStarText = styled(CardStarText)``;

export const HotelDescription = styled.div`
  color: ${theme.palette.neutral};
  padding: ${theme.spacing.gutter}px 0;
  font-size: ${theme.fonts.sizes.normal}px;
  line-height: 17px;
`;

export const HotelHighlights = styled(CardHighlights)`
  border: 0;
  margin: ${theme.spacing.gutter}px 0;
  padding: 0;

  ${props => props.theme.breakpoints.tablet`
        display: block;
    `}
`;

export const HotelHighlight = styled(CardHighlight)`
  :last-child {
    margin-bottom: 0;
  }

  ${props => props.theme.breakpoints.tablet`
        flex: unset;
        width: unset;
    `}
`;

export const HotelGallery = styled.div`
  margin: 0 0 ${theme.spacing.gutter * 2.5}px;

  ${props => props.theme.breakpoints.tablet`
   margin-bottom: ${theme.spacing.gutter * 4}px;
   margin-right: ${theme.spacing.gutter * 3.6}px;
   margin-left: ${theme.spacing.gutter * 2}px;
  `}
`;

export const FixedSlider = styled(Slider)`
  ${props => props.theme.breakpoints.tablet`
    .slick-list, .slick-track, .slick-slide, .slick-slide > div {
      height: 100%;
    }

    .slick-slide img {
      height: 100%;
      object-fit: cover;
    }
  `}
`;

export const MainSlider = styled(FixedSlider)`
  ${props => props.theme.breakpoints.tablet`
    height: 500px;
  `}
`;

export const NavSlider = styled(FixedSlider)`
  padding: 0;

  .slick-track {
    margin: 0;
  }
  .slick-slide {
    padding: ${theme.spacing.gutter / 2}px;

    :first-child {
      margin-left: 0;
    }

    :last-child {
      margin-right: 0;
    }
  }

  ${props => props.theme.breakpoints.tablet`
    height: 100px;
  `}
`;

export const HotelInfo = styled.div`
  margin: 0 ${theme.spacing.gutter * 2}px ${theme.spacing.gutter * 4.6}px;

  ${props => props.theme.breakpoints.tablet`
    margin-bottom: ${theme.spacing.gutter * 8}px;
    padding-right: 5%;
  `}
`;

export const StyledRooms = styled(Rooms)``;

export const HotelWrapper = styled.div`
  ${props => props.theme.breakpoints.tablet`
  `}
`;
