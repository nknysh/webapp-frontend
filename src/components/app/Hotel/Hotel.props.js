import PropTypes from 'prop-types';

import { noop } from 'utils';

export const propTypes = {
  onRoomAdd: PropTypes.func,
  onRoomRemove: PropTypes.func,
  selectedRooms: PropTypes.object,
};

export const defaultProps = {
  onRoomAdd: noop,
  onRoomRemove: noop,
  selectedRooms: {},
};
