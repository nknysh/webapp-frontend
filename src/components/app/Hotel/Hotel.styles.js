import styled from 'styled-components';

import { Slider, Card } from 'components/elements';
import Rooms from 'containers/Rooms';

import {
  CardRating,
  CardStarRating,
  CardStar,
  CardStarText,
  CardSecondaryRating,
  CardHighlights,
  CardHighlight,
} from 'containers/SearchResult/SearchResult.styles';

import theme from 'styles/theme';
import breakpoints from 'styles/breakpoints';

export const StyledHotel = styled.div``;

export const HotelDetails = styled.div``;

export const HotelDetailsRow = styled.div`
  display: block;

  ${breakpoints.tablet`
        display: flex;
    `}
`;

const HotelDetailsColumn = styled.div`
  margin: 0;

  :not(:empty) {
    padding: ${theme.gutter / 2}px 0;
    border-bottom: 1px solid ${theme.borders.default};
  }

  ${breakpoints.tablet`
        padding: ${theme.gutter / 2}px 0;
        border-bottom: 1px solid ${theme.borders.default};
        flex: 1;
    `}
`;

export const HotelDetailsColumnLeft = styled(HotelDetailsColumn)`
  ${breakpoints.tablet`
    margin-right: ${theme.gutter}px;
        max-width: 75%;
    `}
`;

export const HotelDetailsColumnRight = styled(HotelDetailsColumn)`
  ${breakpoints.tablet`
  margin-left: ${theme.gutter}px;
        max-width: 25%;
    `}
`;

export const HotelName = styled(Card.Title)`
  border-bottom: 0;
  margin: 0 0 ${theme.gutter / 2}px;
  padding: 0;
`;

export const HotelRegion = styled.h3`
  font-size: ${theme.fonts.sizes.normal}px;
  color: ${theme.neutral};
  text-transform: uppercase;
  font-weight: ${theme.fonts.normal};
  margin: 0 0 ${theme.gutter}px;
  padding: 0;
`;

export const HotelRating = styled(CardRating)`
  border: 0;
  padding: 0;

  :last-child {
    margin-bottom: 0;
  }

  ${breakpoints.tablet`
        display: block;
    `}
`;

export const HotelStarRating = styled(CardStarRating)`
  border: 0;

  :last-child {
    margin-bottom: 0;
  }

  ${breakpoints.tablet`
        margin: ${theme.gutter / 2}px 0 ${theme.gutter}px;
    `}
`;

export const HotelStar = styled(CardStar)`
  font-size: 20px !important;
`;

export const HotelStarText = styled(CardStarText)``;

export const HotelSecondaryRating = styled(CardSecondaryRating)`
  text-align: right;

  :last-child {
    margin-bottom: 0;
  }

  ${breakpoints.tablet`
        width: unset;
        text-align: unset;
        font-weight: ${theme.fonts.normal};
    `}
`;

export const HotelDescription = styled.div`
  color: ${theme.neutral};
  padding: ${theme.gutter}px 0;
  font-size: ${theme.fonts.sizes.normal}px;
  line-height: 17px;
`;

export const HotelHighlights = styled(CardHighlights)`
  border: 0;
  margin: ${theme.gutter}px 0;
  padding: 0;

  ${breakpoints.tablet`
        display: block;
    `}
`;

export const HotelHighlight = styled(CardHighlight)`
  :last-child {
    margin-bottom: 0;
  }

  ${breakpoints.tablet`
        flex: unset;
        width: unset;
    `}
`;

export const HotelGallery = styled.div`
  margin: 0 0 ${theme.gutter * 2.5}px;

  ${breakpoints.tablet`
   margin-bottom: ${theme.gutter * 4}px;
   margin-right: ${theme.gutter * 3.6}px;
   margin-left: ${theme.gutter * 2}px;
  `}
`;

export const FixedSlider = styled(Slider)`
  ${breakpoints.tablet`
    .slick-list, .slick-track, .slick-slide, .slick-slide > div {
      height: 100%;
    }

    .slick-slide img {
      height: 100%;
      object-fit: cover;
    }
  `}
`;

export const MainSlider = styled(FixedSlider)`
  ${breakpoints.tablet`
    height: 500px;
  `}
`;

export const NavSlider = styled(FixedSlider)`
  padding: 0;

  .slick-track {
    margin: 0;
  }
  .slick-slide {
    padding: ${theme.gutter / 2}px;

    :first-child {
      margin-left: 0;
    }

    :last-child {
      margin-right: 0;
    }
  }

  ${breakpoints.tablet`
    height: 100px;
  `}
`;

export const HotelInfo = styled.div`
  margin: 0 ${theme.gutter * 2}px ${theme.gutter * 4.6}px;

  ${breakpoints.tablet`
    margin-bottom: ${theme.gutter * 8}px;
  `}
`;

export const StyledRooms = styled(Rooms)``;

export const HotelWrapper = styled.div`
  ${breakpoints.tablet`
  `}
`;
