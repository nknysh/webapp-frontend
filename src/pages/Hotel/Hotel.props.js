import PropTypes from 'prop-types';

export const propTypes = {
  match: PropTypes.shape({
    params: PropTypes.object,
  }),
};

export const defaultProps = {
  match: {
    params: {},
  },
};
