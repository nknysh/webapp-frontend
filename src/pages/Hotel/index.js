import { lazy } from 'react';

export { default } from './Hotel';

export const AsyncHotel = lazy(() => import('./Hotel'));
