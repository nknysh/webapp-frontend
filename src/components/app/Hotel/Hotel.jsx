import React, { useRef } from 'react';
import { map, values } from 'ramda';
import { isNilOrEmpty } from 'ramda-adjunct';
import { useTranslation } from 'react-i18next';

import { propTypes, defaultProps } from './Hotel.props';
import {
  HotelDescription,
  HotelDetails,
  HotelDetailsColumnLeft,
  HotelDetailsColumnRight,
  HotelDetailsRow,
  HotelGallery,
  HotelHighlight,
  HotelHighlights,
  HotelInfo,
  HotelName,
  HotelRating,
  HotelRegion,
  HotelSecondaryRating,
  HotelStar,
  HotelStarRating,
  HotelStarText,
  MainSlider,
  NavSlider,
  StyledHotel,
  StyledRooms,
} from './Hotel.styles';

// eslint-disable-next-line
const renderImage = ({ displayName, url }) => <img key={url} src={url} alt={displayName} />;
const renderFeature = value => <HotelHighlight key={value}>{value}</HotelHighlight>;

export const Hotel = ({
  id,
  amenities,
  description,
  name,
  photos,
  region,
  starRating,
  suitableForHoneymooners,
  ...props
}) => {
  const { t } = useTranslation();

  const sliderMain = useRef(null);
  const sliderNav = useRef(null);

  const renderedPhotos = values(map(renderImage, photos));

  return (
    <StyledHotel {...props}>
      <HotelDetails>
        {!isNilOrEmpty(photos) && (
          <HotelGallery>
            <MainSlider asNavFor={sliderNav} centerMode={false} fade={true} ref={sliderMain} slidesToShow={1}>
              {renderedPhotos}
            </MainSlider>
            <NavSlider
              asNavFor={sliderMain}
              centerMode={false}
              focusOnSelect={true}
              infinite={false}
              ref={sliderNav}
              slidesToShow={6}
              arrows={true}
            >
              {renderedPhotos}
            </NavSlider>
          </HotelGallery>
        )}
        <HotelInfo>
          <HotelDetailsRow>
            <HotelDetailsColumnLeft>
              <HotelName>{name}</HotelName>
              {region && <HotelRegion>{region}</HotelRegion>}
            </HotelDetailsColumnLeft>
            <HotelDetailsColumnRight>
              <HotelRating>
                <HotelStarRating>
                  <HotelStar>star</HotelStar>{' '}
                  <HotelStarText>
                    {starRating} {t('star')}
                  </HotelStarText>
                </HotelStarRating>
                {suitableForHoneymooners && (
                  <HotelSecondaryRating>{t('taglines.suitableHoneymoon')}</HotelSecondaryRating>
                )}
              </HotelRating>
            </HotelDetailsColumnRight>
          </HotelDetailsRow>
          <HotelDetailsRow>
            <HotelDetailsColumnLeft>
              <HotelDescription>{description}</HotelDescription>
            </HotelDetailsColumnLeft>
            <HotelDetailsColumnRight>
              {!isNilOrEmpty(amenities) && <HotelHighlights>{map(renderFeature, amenities)}</HotelHighlights>}
            </HotelDetailsColumnRight>
          </HotelDetailsRow>
        </HotelInfo>
        <StyledRooms hotelUuid={id} />
      </HotelDetails>
    </StyledHotel>
  );
};

Hotel.propTypes = propTypes;
Hotel.defaultProps = defaultProps;

export default Hotel;
