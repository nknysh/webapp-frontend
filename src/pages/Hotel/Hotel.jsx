import React from 'react';

import { HotelContainer } from 'containers';

import { propTypes, defaultProps } from './Hotel.props';
import {} from './Hotel.styles';

export const Hotel = ({
  match: {
    params: { id },
  },
}) => <HotelContainer id={id} />;

Hotel.propTypes = propTypes;
Hotel.defaultProps = defaultProps;

export default Hotel;
