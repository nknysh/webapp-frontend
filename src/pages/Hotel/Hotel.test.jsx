import React from 'react';

import { Hotel } from './Hotel';

const getComponent = props => shallow(<Hotel {...props} />);

describe('<Hotel />', () => {
  describe('render', () => {
    it('matches snapshot', () => {
      expect(getComponent()).toMatchSnapshot();
    });
  });
});
