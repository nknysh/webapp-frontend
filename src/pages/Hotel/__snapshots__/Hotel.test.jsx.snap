// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Hotel /> render matches snapshot 1`] = `<Connect(withRouter(withUser())) />`;
